blockchain/types/src/block_metadata.rs

```diff
@@ -129,3 +157,70 @@ impl BlockMetadata {
         &self.signature
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::QuorumSetMember;
+    use alloc::vec;
+    use mc_blockchain_test_utils::test_node_id;
+    use mc_crypto_digestible::MerlinTranscript;
+
+    /// Metadata contents used in block version 3
+    #[derive(Clone, Deserialize, Digestible, Display, Eq, Message, PartialEq, Serialize)]
+    #[digestible(name = "BlockMetadataContents")]
+    struct BlockMetadataContentsV3 {
+        /// The Block ID.
+        #[prost(message, required, tag = 1)]
+        block_id: BlockID,
+
+        /// Quorum set configuration at the time of externalization.
+        #[prost(message, required, tag = 2)]
+        quorum_set: QuorumSet,
+
+        /// IAS report for the enclave which generated the signature.
+        #[prost(message, required, tag = 3)]
+        verification_report: VerificationReport,
+
+        /// Responder ID of the consensus node that externalized this block.
+        #[prost(message, required, tag = 4)]
+        responder_id: ResponderId,
+    }
+
+    #[test]
+    fn metadata_contents_version_3_works_with_version_4() {
+        let quorum_set = QuorumSet::new(
+            2,
+            vec![
+                QuorumSetMember::Node(test_node_id(9)),
+                QuorumSetMember::Node(test_node_id(8)),
+                QuorumSetMember::Node(test_node_id(7)),
+            ],
+        );
+
+        let block_v3 = BlockMetadataContentsV3 {
+            block_id: BlockID([1; 32]),
+            quorum_set: quorum_set.clone(),
+            verification_report: VerificationReport::default(),
```

Comment:
Would be nice to test with something that is not all zeros, to represent a more real-world scenario.

Reply:
Test case has been updated to use `run_with_several_seeds` and `make_verification_report()`